src/stk/ea/mutation/mutators/molecule/random_smarts.py

```diff
@@ -0,0 +1,121 @@
+"""
+Random Smarts
+=============
+
+"""
+
+import numpy as np
+
+
+from .mutator import MoleculeMutator
+from ...records import MutationRecord
+from ....molecule_records import MoleculeRecord
+from rdkit import Chem
+from .....molecular import BuildingBlock
+
+
+class RandomSmarts(MoleculeMutator):
+    """
+    Substitutes functional groups within building blocks.
+
+    This mutator takes a :class:`ConstructedMolecule` and substitutes
+    the existing building blocks with ones containing
+    new functionalities, as specified in the SMARTS strings.
+    Atoms in functional groups cannot be adjusted as these are used to
+    construct the :class: `ConstructedMolecule`, and will
+    result in an error being raised.
```

Comment:
Is this the case? I would have thought that since we have the `replacement_fgs` parameter, it should be ok?

Reply:
nah its good now I think (though I think its probably worth adding an example to aid the expalanation as per my other comment